lib/OpenQA/Git.pm

```diff
@@ -30,9 +30,11 @@ sub _validate_attributes ($self) {
 
 sub _run_cmd ($self, $args, $options = {}) {
     my $include_git_path = $options->{include_git_path} // 1;
-    my $ssh_batchmode = $options->{ssh_batchmode} // 0;
+    #warn(">>>> $options");
+    my $batchmode = $options->{batchmode} // 0;
     my @cmd;
-    push @cmd, 'env', 'GIT_SSH_COMMAND=ssh -oBatchMode=yes' if $ssh_batchmode;
+    push @cmd, 'env', 'GIT_SSH_COMMAND=ssh -oBatchMode=yes' if $batchmode;
```

Comment:
Why did you not use `GIT_ASKPASS` and `GIT_TERMINAL_PROMPT` like suggested in the ticket?
Waiting for a timing out command seems way more complicated than just telling git to not prompt in the first place.

Reply:
I was thinking that with the `timeout` we gonna have a general solution when the _run_cmd (or more specifically  run_cmd_with_log_return_error) encounter similar issue. 
GIT_ASKPASS and GIT_TERMINAL_PROMPT make the logic more complicated IMO but also provide a better error as I didnt find a way to catch the prompt in the $stdout somehow. 
Maybe both can be applied at the end.

Reply:
I'm not sure what's complicated about using two env vars. It's as simple as it gets?